Update simpleBluez

@mh4x0f As you know `zmkBATx` is on nixpkgs and it was recently (automatically) created a PR for updating `simpleBluez` to the version `0.8.1`: https://github.com/NixOS/nixpkgs/pull/354266

This breaks `zmkBATx` (the only package using `simpleBluez`).

<details>
  <summary>Build logs</summary>

  ```
  Running phase: qmakePrePhase
@nix { "action": "setPhase", "phase": "qmakePrePhase" }
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/1n7iyvvn84d1ciscb1bpm8djywkv1h65-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
QMAKEPATH=/nix/store/98w652armi076961wkf75vfa7mp215z7-qtbase-6.8.0-dev:/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0:/nix/store/a8lkfn0g4ywgw8dvap00mph8z3lk9rwq-qtconnectivity-6.8.0-dev:/nix/store/b49jphj177y7v5hffqcldaa2vi77mpkc-qtdeclarative-6.8.0-dev:/nix/store/sjzkqca2fi9448df1k557f6dwjl63xlx-qtlanguageserver-6.8.0-dev:/nix/store/vqyfp249cmwlcr8rqkw6r7b71888yfbk-qtshadertools-6.8.0-dev:/nix/store/857cwg5hg63bq9kqppl8182ag09kc2si-qtdeclarative-6.8.0
qmake PREFIX=/nix/store/wmmc8nwb4s415wncjwikswwfp20px0v4-zmkBATx-1.0.1 NIX_OUTPUT_OUT=/nix/store/wmmc8nwb4s415wncjwikswwfp20px0v4-zmkBATx-1.0.1 NIX_OUTPUT_DEV=/nix/store/wmmc8nwb4s415wncjwikswwfp20px0v4-zmkBATx-1.0.1 NIX_OUTPUT_BIN=/nix/store/wmmc8nwb4s415wncjwikswwfp20px0v4-zmkBATx-1.0.1 NIX_OUTPUT_QML=/nix/store/wmmc8nwb4s415wncjwikswwfp20px0v4-zmkBATx-1.0.1/lib/qt-6/qml NIX_OUTPUT_PLUGIN=/nix/store/wmmc8nwb4s415wncjwikswwfp20px0v4-zmkBATx-1.0.1/lib/qt-6/plugins
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
Info: creating stash file /build/source/.qmake.stash
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
qmake: enabled parallel building
qmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j24 SHELL=/nix/store/mm0pa3z7kk6jh1i9rkxqxjqmd8h1qpxf-bash-5.2p37/bin/bash
/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/libexec/uic src/forms/mainwindow.ui -o src/forms/ui_mainwindow.h
/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/libexec/uic src/forms/about.ui -o src/forms/ui_about.h
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/main.o src/main.cpp
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/appSettings.o src/appSettings.cpp
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/bleinterfacebluez.o src/bleinterfacebluez.cpp
/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/libexec/rcc -name resources resources.qrc -o build/x86_64/qrc_resources.cpp
g++ -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -dM -E -o build/x86_64/moc_predefs.h /nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/features/data/dummy.cpp
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/mainwindow.o src/mainwindow.cpp
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/about.o src/about.cpp
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/qrc_resources.o build/x86_64/qrc_resources.cpp
/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/libexec/moc -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB --include /build/source/build/x86_64/moc_predefs.h -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -I/build/source -I/build/source/src -I/build/source/src/thirdparty -I/build/source/src/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -I/nix/store/0wr6hjdlrrnxxzshnx0pfbkflnhc1kxv-libxml2-2.13.4-dev/include -I/nix/store/776ai37hykgzfg2gafvzsaf7bvk7ym1w-libxslt-1.1.42-dev/include -I/nix/store/xm6d1ig9ff7zkid1gvzsbig45m2pnlaz-openssl-3.3.2-dev/include -I/nix/store/khmgspvzvab2593zsdv165wn55fnql96-sqlite-3.46.1-dev/include -I/nix/store/cir85maag3aj3sddq35y59m0wmcr50wk-zlib-1.3.1-dev/include -I/nix/store/sv5fvv7pimim1z48yyy4fs0amr93qg4q-harfbuzz-9.0.0-dev/include -I/nix/store/8rnglxqis2dp82571wxp8ahbbn0xclj0-graphite2-1.3.14-dev/include -I/nix/store/lqafrbq3phvvvpw3fahzn7vbvl01xibw-icu4c-74.2-dev/include -I/nix/store/gdccq1dnaqibc14v8dhj09mpimbpsr3q-libjpeg-turbo-3.0.3-dev/include -I/nix/store/w4fjaz34v3w4wipnh3inbdpzss43cx59-libpng-apng-1.6.43-dev/include -I/nix/store/bq52gkcn78743zgbrcjdqvizvah6sid8-pcre2-10.44-dev/include -I/nix/store/vdysymr503cz6dnk8qmc446z36yx80l9-pcre-8.45-dev/include -I/nix/store/h9185p3s0q345sd8m2vsgbhij7n74iif-zstd-1.5.6-dev/include -I/nix/store/f0m71kd8nzripsnw8l3w27vs78czamjp-libb2-0.98.1/include -I/nix/store/rbrg4l9nf6b2fic74hgrb721kjgd1kwm-md4c-0.5.2-dev/include -I/nix/store/dsgnd4bpc0c9wpqp4z966hhhqvxkii0n-double-conversion-3.3.0/include -I/nix/store/sk5s9938n7qqfm1nj95kgqja8gm06xhv-libproxy-0.5.9-dev/include -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include -I/nix/store/vl8sjsq5mkb9vgwy75divdjv4yf9fidf-expat-2.6.3-dev/include -I/nix/store/1hhi03gg5jdb2fa5gb9ncklziczax741-glib-2.82.1-dev/include -I/nix/store/rhl22j48423aazmhjkwndpr1syzq5l46-libffi-3.4.6-dev/include -I/nix/store/rv0w5c2wp24a2s7yfhqqk7w23k2j90kx-gettext-0.21.1/include -I/nix/store/iap3ggsrjhgwn2p2n5y6qdv1r4m43gzq-glibc-iconv-2.40/include -I/nix/store/q8axfkpn2npzyp9gggjy1wk19kpf6ai1-unixODBC-2.3.12/include -I/nix/store/qy7rfm4gzga4r12s9l08g7b9ca8r3n8a-systemd-256.7-dev/include -I/nix/store/hl9qy8sbq6a0xykg4a83v9rprc6viizk-util-linux-2.39.4-dev/include -I/nix/store/mlzk6bqinkcz02npbr7vcnq29pzqpd95-mtdev-1.1.7/include -I/nix/store/dvx8w800mjgwc1k98ibn6hlx1xwv5hgb-lksctp-tools-1.0.21/include -I/nix/store/cz56nkdpazr28rzrcg52aggfhrhp3lb1-libselinux-3.7-dev/include -I/nix/store/axbhbim2q1mlqc7rz3aiqwcmmbbp3n13-libsepol-3.7-dev/include -I/nix/store/jqxnplic8zbc3zj8wxcvk70ddcpilfjx-lttng-ust-2.13.8-dev/include -I/nix/store/83lwc3rwii81zbw2yxl32qg9brc559jy-liburcu-0.14.1-dev/include -I/nix/store/l8khs1jbgr61fc58798199vikgcmac2w-vulkan-headers-1.3.296.0/include -I/nix/store/igz98qsl5g5c1s0i6ghlrsjwracbw05r-libthai-0.1.29-dev/include -I/nix/store/lp2mk0k2qx41y2hln8vdbi003i434vr3-libdrm-2.4.123-dev/include -I/nix/store/hmlwqigbn9f7gnlzl97qv51xasg4ijxw-libdatrie-2019-12-20-dev/include -I/nix/store/s94mkdlqznq2f8p8aqkvkpdp15hkx9jv-systemd-minimal-libs-256.7-dev/include -I/nix/store/6k3dxq8hhb5l278n6icl79njcgykjcy5-fontconfig-2.15.0-dev/include -I/nix/store/l1shggh8syl6kra49whf8mpjhhd73360-freetype-2.13.3-dev/include -I/nix/store/z941y5ja7n937hw7y9m35ywq58812c0i-bzip2-1.0.8-dev/include -I/nix/store/w6qhkwgdiwd0krzafw26wrn1f6m8xz13-brotli-1.1.0-dev/include -I/nix/store/xrjakm88sn8q139d5m5jknf7yhyavb53-libX11-1.8.10-dev/include -I/nix/store/dx8n1ab4h1gvzn17kpbyizgjr8hnsmf1-xorgproto-2024.1/include -I/nix/store/l77hg7iaih1i81fqb3lypq2zci229gmy-libxcb-1.17.0-dev/include -I/nix/store/0hqmx97nqc92nyq2zg4ana7lhlpwp74w-libXcomposite-0.4.6-dev/include -I/nix/store/i0bm1dmc1s9bkxgnlk4jdrq0ipdsjh27-libXfixes-6.0.1-dev/include -I/nix/store/sv0aaycvf23n7yynvfaisb77srbbyvww-libXext-1.3.6-dev/include -I/nix/store/2r9mscvzk9vfa3whzfzhlk20hbi243ws-libXau-1.0.11-dev/include -I/nix/store/gjkq07qj8rhzqb8qvhxds0nxi8ff3pri-libXi-1.8.2-dev/include -I/nix/store/akzcx4v9z316h8ddbzflysryw3lrpyzl-libXrender-0.9.11-dev/include -I/nix/store/8rma94r6y5kqjkiqmzbsahapq109sa4x-libxkbcommon-1.7.0-dev/include -I/nix/store/ppmd42917bcf7gxivpnnfm6ljw4w585y-xcb-util-0.4.1-dev/include -I/nix/store/a6qkhv569wg6jrqk1cpzbsqin8vziaj0-xcb-util-image-0.4.1-dev/include -I/nix/store/3pzkjb62iw7q53rpn336rdgbkkc23qqw-xcb-util-keysyms-0.4.1-dev/include -I/nix/store/iyyvlrb4bzwmpma2cb9sd5fpy89b1mdm-xcb-util-renderutil-0.3.10-dev/include -I/nix/store/l2pfkml9aarp1dgy81m3cj36sjjh7k15-xcb-util-wm-0.4.2-dev/include -I/nix/store/8q7dydw2i9rpypri4xx4cdlgv4987sqz-libXdmcp-1.1.5-dev/include -I/nix/store/414cxmwgzky1vkm6kc1drvr2w62r6l4x-libXtst-1.2.5/include -I/nix/store/324gbcphxspmwdzq544g12ccdc3qw6q0-xcb-util-cursor-0.1.5-dev/include -I/nix/store/p0gpmxihb88q4i72f8iwjfdi9vk7a2gy-libepoxy-1.5.10-dev/include -I/nix/store/hbg7illckfcpcav62gafgcdibjpjyqfm-libglvnd-1.7.0-dev/include -I/nix/store/gpk9rnndipq776hbk49jv8933axi9qm9-cups-2.4.11-dev/include -I/nix/store/g5pla62i42xr0pk1qlab06qnalriw6wh-gmp-with-cxx-6.3.0-dev/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include -I/nix/store/kwb0rygdpl9fm3c6nm168cwgi7gp2n2v-qtlanguageserver-6.8.0/include -I/nix/store/46kfcn62g55nhbx91f5q6wwrvf2gghhq-qtshadertools-6.8.0/include -I/nix/store/857cwg5hg63bq9kqppl8182ag09kc2si-qtdeclarative-6.8.0/include -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include -I/nix/store/fkxbiagkqjfabq4fcac62yz5wzak4390-simpleBluez-0.8.1-dev/include -I/nix/store/1yiaji6dnh07b8v17chscr2dgf7ffmp3-simpleDBus-0.7.3-dev/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0 -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0/x86_64-unknown-linux-gnu -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0/backward -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/lib/gcc/x86_64-unknown-linux-gnu/13.3.0/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/lib/gcc/x86_64-unknown-linux-gnu/13.3.0/include-fixed -I/nix/store/932dj5qwfzck90mnvqpd1f9hjqznaqdj-glibc-2.40-36-dev/include src/mainwindow.h -o build/x86_64/moc_mainwindow.cpp
/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/libexec/moc -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB --include /build/source/build/x86_64/moc_predefs.h -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -I/build/source -I/build/source/src -I/build/source/src/thirdparty -I/build/source/src/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -I/nix/store/0wr6hjdlrrnxxzshnx0pfbkflnhc1kxv-libxml2-2.13.4-dev/include -I/nix/store/776ai37hykgzfg2gafvzsaf7bvk7ym1w-libxslt-1.1.42-dev/include -I/nix/store/xm6d1ig9ff7zkid1gvzsbig45m2pnlaz-openssl-3.3.2-dev/include -I/nix/store/khmgspvzvab2593zsdv165wn55fnql96-sqlite-3.46.1-dev/include -I/nix/store/cir85maag3aj3sddq35y59m0wmcr50wk-zlib-1.3.1-dev/include -I/nix/store/sv5fvv7pimim1z48yyy4fs0amr93qg4q-harfbuzz-9.0.0-dev/include -I/nix/store/8rnglxqis2dp82571wxp8ahbbn0xclj0-graphite2-1.3.14-dev/include -I/nix/store/lqafrbq3phvvvpw3fahzn7vbvl01xibw-icu4c-74.2-dev/include -I/nix/store/gdccq1dnaqibc14v8dhj09mpimbpsr3q-libjpeg-turbo-3.0.3-dev/include -I/nix/store/w4fjaz34v3w4wipnh3inbdpzss43cx59-libpng-apng-1.6.43-dev/include -I/nix/store/bq52gkcn78743zgbrcjdqvizvah6sid8-pcre2-10.44-dev/include -I/nix/store/vdysymr503cz6dnk8qmc446z36yx80l9-pcre-8.45-dev/include -I/nix/store/h9185p3s0q345sd8m2vsgbhij7n74iif-zstd-1.5.6-dev/include -I/nix/store/f0m71kd8nzripsnw8l3w27vs78czamjp-libb2-0.98.1/include -I/nix/store/rbrg4l9nf6b2fic74hgrb721kjgd1kwm-md4c-0.5.2-dev/include -I/nix/store/dsgnd4bpc0c9wpqp4z966hhhqvxkii0n-double-conversion-3.3.0/include -I/nix/store/sk5s9938n7qqfm1nj95kgqja8gm06xhv-libproxy-0.5.9-dev/include -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include -I/nix/store/vl8sjsq5mkb9vgwy75divdjv4yf9fidf-expat-2.6.3-dev/include -I/nix/store/1hhi03gg5jdb2fa5gb9ncklziczax741-glib-2.82.1-dev/include -I/nix/store/rhl22j48423aazmhjkwndpr1syzq5l46-libffi-3.4.6-dev/include -I/nix/store/rv0w5c2wp24a2s7yfhqqk7w23k2j90kx-gettext-0.21.1/include -I/nix/store/iap3ggsrjhgwn2p2n5y6qdv1r4m43gzq-glibc-iconv-2.40/include -I/nix/store/q8axfkpn2npzyp9gggjy1wk19kpf6ai1-unixODBC-2.3.12/include -I/nix/store/qy7rfm4gzga4r12s9l08g7b9ca8r3n8a-systemd-256.7-dev/include -I/nix/store/hl9qy8sbq6a0xykg4a83v9rprc6viizk-util-linux-2.39.4-dev/include -I/nix/store/mlzk6bqinkcz02npbr7vcnq29pzqpd95-mtdev-1.1.7/include -I/nix/store/dvx8w800mjgwc1k98ibn6hlx1xwv5hgb-lksctp-tools-1.0.21/include -I/nix/store/cz56nkdpazr28rzrcg52aggfhrhp3lb1-libselinux-3.7-dev/include -I/nix/store/axbhbim2q1mlqc7rz3aiqwcmmbbp3n13-libsepol-3.7-dev/include -I/nix/store/jqxnplic8zbc3zj8wxcvk70ddcpilfjx-lttng-ust-2.13.8-dev/include -I/nix/store/83lwc3rwii81zbw2yxl32qg9brc559jy-liburcu-0.14.1-dev/include -I/nix/store/l8khs1jbgr61fc58798199vikgcmac2w-vulkan-headers-1.3.296.0/include -I/nix/store/igz98qsl5g5c1s0i6ghlrsjwracbw05r-libthai-0.1.29-dev/include -I/nix/store/lp2mk0k2qx41y2hln8vdbi003i434vr3-libdrm-2.4.123-dev/include -I/nix/store/hmlwqigbn9f7gnlzl97qv51xasg4ijxw-libdatrie-2019-12-20-dev/include -I/nix/store/s94mkdlqznq2f8p8aqkvkpdp15hkx9jv-systemd-minimal-libs-256.7-dev/include -I/nix/store/6k3dxq8hhb5l278n6icl79njcgykjcy5-fontconfig-2.15.0-dev/include -I/nix/store/l1shggh8syl6kra49whf8mpjhhd73360-freetype-2.13.3-dev/include -I/nix/store/z941y5ja7n937hw7y9m35ywq58812c0i-bzip2-1.0.8-dev/include -I/nix/store/w6qhkwgdiwd0krzafw26wrn1f6m8xz13-brotli-1.1.0-dev/include -I/nix/store/xrjakm88sn8q139d5m5jknf7yhyavb53-libX11-1.8.10-dev/include -I/nix/store/dx8n1ab4h1gvzn17kpbyizgjr8hnsmf1-xorgproto-2024.1/include -I/nix/store/l77hg7iaih1i81fqb3lypq2zci229gmy-libxcb-1.17.0-dev/include -I/nix/store/0hqmx97nqc92nyq2zg4ana7lhlpwp74w-libXcomposite-0.4.6-dev/include -I/nix/store/i0bm1dmc1s9bkxgnlk4jdrq0ipdsjh27-libXfixes-6.0.1-dev/include -I/nix/store/sv0aaycvf23n7yynvfaisb77srbbyvww-libXext-1.3.6-dev/include -I/nix/store/2r9mscvzk9vfa3whzfzhlk20hbi243ws-libXau-1.0.11-dev/include -I/nix/store/gjkq07qj8rhzqb8qvhxds0nxi8ff3pri-libXi-1.8.2-dev/include -I/nix/store/akzcx4v9z316h8ddbzflysryw3lrpyzl-libXrender-0.9.11-dev/include -I/nix/store/8rma94r6y5kqjkiqmzbsahapq109sa4x-libxkbcommon-1.7.0-dev/include -I/nix/store/ppmd42917bcf7gxivpnnfm6ljw4w585y-xcb-util-0.4.1-dev/include -I/nix/store/a6qkhv569wg6jrqk1cpzbsqin8vziaj0-xcb-util-image-0.4.1-dev/include -I/nix/store/3pzkjb62iw7q53rpn336rdgbkkc23qqw-xcb-util-keysyms-0.4.1-dev/include -I/nix/store/iyyvlrb4bzwmpma2cb9sd5fpy89b1mdm-xcb-util-renderutil-0.3.10-dev/include -I/nix/store/l2pfkml9aarp1dgy81m3cj36sjjh7k15-xcb-util-wm-0.4.2-dev/include -I/nix/store/8q7dydw2i9rpypri4xx4cdlgv4987sqz-libXdmcp-1.1.5-dev/include -I/nix/store/414cxmwgzky1vkm6kc1drvr2w62r6l4x-libXtst-1.2.5/include -I/nix/store/324gbcphxspmwdzq544g12ccdc3qw6q0-xcb-util-cursor-0.1.5-dev/include -I/nix/store/p0gpmxihb88q4i72f8iwjfdi9vk7a2gy-libepoxy-1.5.10-dev/include -I/nix/store/hbg7illckfcpcav62gafgcdibjpjyqfm-libglvnd-1.7.0-dev/include -I/nix/store/gpk9rnndipq776hbk49jv8933axi9qm9-cups-2.4.11-dev/include -I/nix/store/g5pla62i42xr0pk1qlab06qnalriw6wh-gmp-with-cxx-6.3.0-dev/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include -I/nix/store/kwb0rygdpl9fm3c6nm168cwgi7gp2n2v-qtlanguageserver-6.8.0/include -I/nix/store/46kfcn62g55nhbx91f5q6wwrvf2gghhq-qtshadertools-6.8.0/include -I/nix/store/857cwg5hg63bq9kqppl8182ag09kc2si-qtdeclarative-6.8.0/include -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include -I/nix/store/fkxbiagkqjfabq4fcac62yz5wzak4390-simpleBluez-0.8.1-dev/include -I/nix/store/1yiaji6dnh07b8v17chscr2dgf7ffmp3-simpleDBus-0.7.3-dev/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0 -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0/x86_64-unknown-linux-gnu -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0/backward -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/lib/gcc/x86_64-unknown-linux-gnu/13.3.0/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/lib/gcc/x86_64-unknown-linux-gnu/13.3.0/include-fixed -I/nix/store/932dj5qwfzck90mnvqpd1f9hjqznaqdj-glibc-2.40-36-dev/include src/about.h -o build/x86_64/moc_about.cpp
/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/libexec/moc -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB --include /build/source/build/x86_64/moc_predefs.h -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -I/build/source -I/build/source/src -I/build/source/src/thirdparty -I/build/source/src/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -I/nix/store/0wr6hjdlrrnxxzshnx0pfbkflnhc1kxv-libxml2-2.13.4-dev/include -I/nix/store/776ai37hykgzfg2gafvzsaf7bvk7ym1w-libxslt-1.1.42-dev/include -I/nix/store/xm6d1ig9ff7zkid1gvzsbig45m2pnlaz-openssl-3.3.2-dev/include -I/nix/store/khmgspvzvab2593zsdv165wn55fnql96-sqlite-3.46.1-dev/include -I/nix/store/cir85maag3aj3sddq35y59m0wmcr50wk-zlib-1.3.1-dev/include -I/nix/store/sv5fvv7pimim1z48yyy4fs0amr93qg4q-harfbuzz-9.0.0-dev/include -I/nix/store/8rnglxqis2dp82571wxp8ahbbn0xclj0-graphite2-1.3.14-dev/include -I/nix/store/lqafrbq3phvvvpw3fahzn7vbvl01xibw-icu4c-74.2-dev/include -I/nix/store/gdccq1dnaqibc14v8dhj09mpimbpsr3q-libjpeg-turbo-3.0.3-dev/include -I/nix/store/w4fjaz34v3w4wipnh3inbdpzss43cx59-libpng-apng-1.6.43-dev/include -I/nix/store/bq52gkcn78743zgbrcjdqvizvah6sid8-pcre2-10.44-dev/include -I/nix/store/vdysymr503cz6dnk8qmc446z36yx80l9-pcre-8.45-dev/include -I/nix/store/h9185p3s0q345sd8m2vsgbhij7n74iif-zstd-1.5.6-dev/include -I/nix/store/f0m71kd8nzripsnw8l3w27vs78czamjp-libb2-0.98.1/include -I/nix/store/rbrg4l9nf6b2fic74hgrb721kjgd1kwm-md4c-0.5.2-dev/include -I/nix/store/dsgnd4bpc0c9wpqp4z966hhhqvxkii0n-double-conversion-3.3.0/include -I/nix/store/sk5s9938n7qqfm1nj95kgqja8gm06xhv-libproxy-0.5.9-dev/include -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include -I/nix/store/vl8sjsq5mkb9vgwy75divdjv4yf9fidf-expat-2.6.3-dev/include -I/nix/store/1hhi03gg5jdb2fa5gb9ncklziczax741-glib-2.82.1-dev/include -I/nix/store/rhl22j48423aazmhjkwndpr1syzq5l46-libffi-3.4.6-dev/include -I/nix/store/rv0w5c2wp24a2s7yfhqqk7w23k2j90kx-gettext-0.21.1/include -I/nix/store/iap3ggsrjhgwn2p2n5y6qdv1r4m43gzq-glibc-iconv-2.40/include -I/nix/store/q8axfkpn2npzyp9gggjy1wk19kpf6ai1-unixODBC-2.3.12/include -I/nix/store/qy7rfm4gzga4r12s9l08g7b9ca8r3n8a-systemd-256.7-dev/include -I/nix/store/hl9qy8sbq6a0xykg4a83v9rprc6viizk-util-linux-2.39.4-dev/include -I/nix/store/mlzk6bqinkcz02npbr7vcnq29pzqpd95-mtdev-1.1.7/include -I/nix/store/dvx8w800mjgwc1k98ibn6hlx1xwv5hgb-lksctp-tools-1.0.21/include -I/nix/store/cz56nkdpazr28rzrcg52aggfhrhp3lb1-libselinux-3.7-dev/include -I/nix/store/axbhbim2q1mlqc7rz3aiqwcmmbbp3n13-libsepol-3.7-dev/include -I/nix/store/jqxnplic8zbc3zj8wxcvk70ddcpilfjx-lttng-ust-2.13.8-dev/include -I/nix/store/83lwc3rwii81zbw2yxl32qg9brc559jy-liburcu-0.14.1-dev/include -I/nix/store/l8khs1jbgr61fc58798199vikgcmac2w-vulkan-headers-1.3.296.0/include -I/nix/store/igz98qsl5g5c1s0i6ghlrsjwracbw05r-libthai-0.1.29-dev/include -I/nix/store/lp2mk0k2qx41y2hln8vdbi003i434vr3-libdrm-2.4.123-dev/include -I/nix/store/hmlwqigbn9f7gnlzl97qv51xasg4ijxw-libdatrie-2019-12-20-dev/include -I/nix/store/s94mkdlqznq2f8p8aqkvkpdp15hkx9jv-systemd-minimal-libs-256.7-dev/include -I/nix/store/6k3dxq8hhb5l278n6icl79njcgykjcy5-fontconfig-2.15.0-dev/include -I/nix/store/l1shggh8syl6kra49whf8mpjhhd73360-freetype-2.13.3-dev/include -I/nix/store/z941y5ja7n937hw7y9m35ywq58812c0i-bzip2-1.0.8-dev/include -I/nix/store/w6qhkwgdiwd0krzafw26wrn1f6m8xz13-brotli-1.1.0-dev/include -I/nix/store/xrjakm88sn8q139d5m5jknf7yhyavb53-libX11-1.8.10-dev/include -I/nix/store/dx8n1ab4h1gvzn17kpbyizgjr8hnsmf1-xorgproto-2024.1/include -I/nix/store/l77hg7iaih1i81fqb3lypq2zci229gmy-libxcb-1.17.0-dev/include -I/nix/store/0hqmx97nqc92nyq2zg4ana7lhlpwp74w-libXcomposite-0.4.6-dev/include -I/nix/store/i0bm1dmc1s9bkxgnlk4jdrq0ipdsjh27-libXfixes-6.0.1-dev/include -I/nix/store/sv0aaycvf23n7yynvfaisb77srbbyvww-libXext-1.3.6-dev/include -I/nix/store/2r9mscvzk9vfa3whzfzhlk20hbi243ws-libXau-1.0.11-dev/include -I/nix/store/gjkq07qj8rhzqb8qvhxds0nxi8ff3pri-libXi-1.8.2-dev/include -I/nix/store/akzcx4v9z316h8ddbzflysryw3lrpyzl-libXrender-0.9.11-dev/include -I/nix/store/8rma94r6y5kqjkiqmzbsahapq109sa4x-libxkbcommon-1.7.0-dev/include -I/nix/store/ppmd42917bcf7gxivpnnfm6ljw4w585y-xcb-util-0.4.1-dev/include -I/nix/store/a6qkhv569wg6jrqk1cpzbsqin8vziaj0-xcb-util-image-0.4.1-dev/include -I/nix/store/3pzkjb62iw7q53rpn336rdgbkkc23qqw-xcb-util-keysyms-0.4.1-dev/include -I/nix/store/iyyvlrb4bzwmpma2cb9sd5fpy89b1mdm-xcb-util-renderutil-0.3.10-dev/include -I/nix/store/l2pfkml9aarp1dgy81m3cj36sjjh7k15-xcb-util-wm-0.4.2-dev/include -I/nix/store/8q7dydw2i9rpypri4xx4cdlgv4987sqz-libXdmcp-1.1.5-dev/include -I/nix/store/414cxmwgzky1vkm6kc1drvr2w62r6l4x-libXtst-1.2.5/include -I/nix/store/324gbcphxspmwdzq544g12ccdc3qw6q0-xcb-util-cursor-0.1.5-dev/include -I/nix/store/p0gpmxihb88q4i72f8iwjfdi9vk7a2gy-libepoxy-1.5.10-dev/include -I/nix/store/hbg7illckfcpcav62gafgcdibjpjyqfm-libglvnd-1.7.0-dev/include -I/nix/store/gpk9rnndipq776hbk49jv8933axi9qm9-cups-2.4.11-dev/include -I/nix/store/g5pla62i42xr0pk1qlab06qnalriw6wh-gmp-with-cxx-6.3.0-dev/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include -I/nix/store/kwb0rygdpl9fm3c6nm168cwgi7gp2n2v-qtlanguageserver-6.8.0/include -I/nix/store/46kfcn62g55nhbx91f5q6wwrvf2gghhq-qtshadertools-6.8.0/include -I/nix/store/857cwg5hg63bq9kqppl8182ag09kc2si-qtdeclarative-6.8.0/include -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include -I/nix/store/fkxbiagkqjfabq4fcac62yz5wzak4390-simpleBluez-0.8.1-dev/include -I/nix/store/1yiaji6dnh07b8v17chscr2dgf7ffmp3-simpleDBus-0.7.3-dev/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0 -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0/x86_64-unknown-linux-gnu -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include/c++/13.3.0/backward -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/lib/gcc/x86_64-unknown-linux-gnu/13.3.0/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/include -I/nix/store/v7dbfh5n7az2lcap0z1cv4jq0bikya8p-gcc-13.3.0/lib/gcc/x86_64-unknown-linux-gnu/13.3.0/include-fixed -I/nix/store/932dj5qwfzck90mnvqpd1f9hjqznaqdj-glibc-2.40-36-dev/include src/bleinterfacebluez.h -o build/x86_64/moc_bleinterfacebluez.cpp
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/moc_bleinterfacebluez.o build/x86_64/moc_bleinterfacebluez.cpp
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/moc_about.o build/x86_64/moc_about.cpp
g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -fPIC -D_REENTRANT -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_BLUETOOTH_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/thirdparty -Isrc/thirdparty -I/nix/store/dzadpismx1a9mn895fcswdl9m683dfg4-dbus-1.14.10-dev/include/dbus-1.0 -I/nix/store/mpwzncl2rnvybrlcn1m6f3ixs08kj2rm-dbus-1.14.10-lib/lib/dbus-1.0/include -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtWidgets -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtGui -I/nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/include/QtBluetooth -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtNetwork -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtDBus -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/include/QtCore -Ibuild/x86_64 -Isrc/forms -I/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/mkspecs/linux-g++ -o build/x86_64/moc_mainwindow.o build/x86_64/moc_mainwindow.cpp
[01m[Ksrc/bleinterfacebluez.cpp:[m[K In function '[01m[Klong unsigned int[01;32m[K hex2dec[m[K(std::string)[m[K':
[01m[Ksrc/bleinterfacebluez.cpp:13:20:[m[K [01;35m[Kwarning: [m[Kcomparison of integer expressions of different signedness: '[01m[Kint[m[K' and '[01m[Kstd::__cxx11::basic_string<char>::size_type[m[K' {aka '[01m[Klong unsigned int[m[K'} [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare-Wsign-compare]8;;[m[K]
   13 |     for (int i=0; [01;35m[Ki<hex.length()[m[K; i++) {
      |                   [01;35m[K~^~~~~~~~~~~~~[m[K
[01m[Ksrc/bleinterfacebluez.cpp:[m[K In member function '[01m[Kvoid BLEInterfaceBluez::[01;32m[KaddDevices[m[K(std::vector<std::shared_ptr<SimpleBluez::Device> >)[m[K':
[01m[Ksrc/bleinterfacebluez.cpp:78:23:[m[K [01;35m[Kwarning: [m[Kcomparison of integer expressions of different signedness: '[01m[Kint[m[K' and '[01m[Kstd::vector<std::shared_ptr<SimpleBluez::Device> >::size_type[m[K' {aka '[01m[Klong unsigned int[m[K'} [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare-Wsign-compare]8;;[m[K]
   78 |     for (int i = 0; [01;35m[Ki < devices.size()[m[K; i++) {
      |                     [01;35m[K~~^~~~~~~~~~~~~~~~[m[K
[01m[Ksrc/bleinterfacebluez.cpp:[m[K In member function '[01m[Kvoid BLEInterfaceBluez::[01;32m[KreadCharacteristicByService[m[K(const std::shared_ptr<SimpleBluez::Service>&)[m[K':
[01m[Ksrc/bleinterfacebluez.cpp:105:42:[m[K [01;35m[Kwarning: [m[Kcatching polymorphic type '[01m[Kclass SimpleBluez::Exception::DescriptorNotFoundException[m[K' by value [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcatch-value=-Wcatch-value=]8;;[m[K]
  105 |         } catch (SimpleBluez::Exception::[01;35m[KDescriptorNotFoundException[m[K) {
      |                                          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Ksrc/bleinterfacebluez.cpp:[m[K In member function '[01m[Kstd::vector<std::shared_ptr<SimpleBluez::Service> > BLEInterfaceBluez::[01;32m[KfindServicesByUuid[m[K(const QBluetoothUuid&)[m[K':
[01m[Ksrc/bleinterfacebluez.cpp:117:27:[m[K [01;35m[Kwarning: [m[Kcomparison of integer expressions of different signedness: '[01m[Kint[m[K' and '[01m[Kstd::vector<std::pair<std::shared_ptr<SimpleBluez::Service>, std::shared_ptr<SimpleBluez::Characteristic> > >::size_type[m[K' {aka '[01m[Klong unsigned int[m[K'} [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare-Wsign-compare]8;;[m[K]
  117 |         for (int i = 0; [01;35m[Ki < m_servicesList.size()[m[K; i++) {
      |                         [01;35m[K~~^~~~~~~~~~~~~~~~~~~~~~~[m[K
g++ -Wl,-O1 -Wl,-rpath,/nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/lib -o bin/zmkbatx  build/x86_64/main.o build/x86_64/about.o build/x86_64/appSettings.o build/x86_64/bleinterfacebluez.o build/x86_64/mainwindow.o build/x86_64/qrc_resources.o build/x86_64/moc_mainwindow.o build/x86_64/moc_about.o build/x86_64/moc_bleinterfacebluez.o   -L/build/source/src/thirdparty/ -lsimplebluez -ldbus-1 /nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/lib/libQt6Widgets.so -lQt6Gui /nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/lib/libQt6Gui.so /nix/store/il5313yg4b804rqaa2dwdvdl7vblcjjl-qtconnectivity-6.8.0/lib/libQt6Bluetooth.so -lQt6Network -lQt6DBus /nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/lib/libQt6Network.so /nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/lib/libQt6DBus.so -lQt6Core /nix/store/065ljrg531nwq9idrmnnh0hc8rn217xf-qtbase-6.8.0/lib/libQt6Core.so -lpthread -lGLX -lOpenGL   
/nix/store/pbixrqd120nnw8a0ygfpm3mnwlr1mbj7-binutils-2.43.1/bin/ld: build/x86_64/bleinterfacebluez.o: in function `BLEInterfaceBluez::readCharacteristicByService(std::shared_ptr<SimpleBluez::Service> const&)':
bleinterfacebluez.cpp:(.text+0x281a): undefined reference to `SimpleBluez::Descriptor::value[abi:cxx11]()'
/nix/store/pbixrqd120nnw8a0ygfpm3mnwlr1mbj7-binutils-2.43.1/bin/ld: bleinterfacebluez.cpp:(.text+0x2c39): undefined reference to `SimpleBluez::Characteristic::read[abi:cxx11]()'
collect2: error: ld returned 1 exit status
make: *** [Makefile:256: bin/zmkbatx] Error 1

  ```
</details>

Do you plan to release a new version?

--- a/src/bleinterfacebluez.cpp
+++ b/src/bleinterfacebluez.cpp
@@ -86,7 +86,7 @@ void BLEInterfaceBluez::addDevices(std::vector<std::shared_ptr<SimpleBluez::Devi
 void BLEInterfaceBluez::readCharacteristicByService(const std::shared_ptr<SimpleBluez::Service> &service )
 {
     for (auto charac : service->characteristics()) {
-        SimpleBluez::ByteArray value = charac->read();
+        std::string value = charac->read();
         auto battery_value = hex2dec(QString::fromStdString(value.c_str()).toUtf8().toHex().toStdString());
         qDebug() << "Characteristic uuid: " << charac->uuid();
         qDebug() << "Characteristic contents: " << battery_value;

--- a/src/thirdparty/simplebluez/Device.h
+++ b/src/thirdparty/simplebluez/Device.h
@@ -29,8 +29,8 @@ class Device : public SimpleDBus::Proxy {
     int16_t rssi();
     int16_t tx_power();
 
-    std::map<uint16_t, std::vector<uint8_t>> manufacturer_data();
-    std::map<std::string, std::vector<uint8_t>> service_data();
+    std::map<uint16_t, ByteArray> manufacturer_data();
+    std::map<std::string, ByteArray> service_data();
 
     bool paired();
     bool connected();

--- a/src/thirdparty/simplebluez/Types.h
+++ b/src/thirdparty/simplebluez/Types.h
@@ -1,9 +1,9 @@
 #pragma once
 
-#include <string>
+#include "external/kvn_bytearray.h"
 
 namespace SimpleBluez {
 
-typedef std::string ByteArray;
+using ByteArray = kvn::bytearray;
 
 }

--- a/src/thirdparty/simplebluez/external/kvn_bytearray.h
+++ b/src/thirdparty/simplebluez/external/kvn_bytearray.h
@@ -1,0 +1,200 @@
+#ifndef KVN_BYTEARRAY_H
+#define KVN_BYTEARRAY_H
+
+#include <cstdint>
+#include <cstring>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <vector>
+
+namespace kvn {
+
+/**
+ * @class bytearray
+ * @brief A class to handle byte arrays and their conversion from/to hex strings.
+ */
+class bytearray {
+  public:
+    /**
+     * @brief Default constructor.
+     */
+    bytearray() = default;
+
+    /**
+     * @brief Constructs byte array from a vector of uint8_t.
+     * @param vec A vector of uint8_t.
+     */
+    bytearray(const std::vector<uint8_t>& vec) : data_(vec) {}
+
+    /**
+     * @brief Constructs byte array from a raw pointer and size.
+     * @param ptr A pointer to uint8_t data.
+     * @param size The size of the data.
+     */
+    bytearray(const uint8_t* ptr, size_t size) : data_(ptr, ptr + size) {}
+
+    /**
+     * @brief Constructs byte array from a std::string.
+     * @param byteArr A string containing byte data.
+     */
+    bytearray(const std::string& byteArr) : data_(byteArr.begin(), byteArr.end()) {}
+
+    /**
+     * @brief Constructs byte array from a C-style string and size.
+     * @param byteArr A C-style string.
+     * @param size The size of the string.
+     */
+    bytearray(const char* byteArr, size_t size) : bytearray(std::string(byteArr, size)) {}
+
+    /**
+     * @brief Constructs byte array from a C-style string.
+     * @param byteArr A C-style string.
+     */
+    bytearray(const char* byteArr) : bytearray(std::string(byteArr)) {}
+
+    /**
+     * @brief Creates a ByteArray from a hex string.
+     *
+     * Case is ignored and the string may have a '0x' hex prefix or not.
+     *
+     * @param hexStr A string containing hex data.
+     * @return A ByteArray object.
+     * @throws std::invalid_argument If the hex string contains non-hexadecimal characters.
+     * @throws std::length_error If the hex string length is not even.
+     */
+    static bytearray fromHex(const std::string& hexStr) {
+        std::string cleanString(hexStr);
+
+        // Check and skip the '0x' prefix if present
+        if (cleanString.size() >= 2 && cleanString.substr(0, 2) == "0x") {
+            cleanString = cleanString.substr(2);
+        }
+
+        size_t size = cleanString.size();
+        if (size % 2 != 0) {
+            throw std::length_error("Hex string length must be even.");
+        }
+
+        bytearray byteArray;
+        byteArray.data_.reserve(size / 2);
+
+        for (size_t i = 0; i < size; i += 2) {
+            uint8_t byte = static_cast<uint8_t>(std::stoi(cleanString.substr(i, 2), nullptr, 16));
+            byteArray.data_.push_back(byte);
+        }
+
+        return byteArray;
+    }
+
+    /**
+     * @overload
+     */
+    static bytearray fromHex(const char* byteArr) { return fromHex(std::string(byteArr)); }
+
+    /**
+     * @overload
+     */
+    static bytearray fromHex(const char* byteArr, size_t size) { return fromHex(std::string(byteArr, size)); }
+
+    /**
+     * @brief Converts the byte array to a lowercase hex string without '0x' prefix.
+     * @param spacing Whether to include spaces between bytes.
+     *
+     * @return A hex string representation of the byte array.
+     */
+    std::string toHex(bool spacing = false) const {
+        std::ostringstream oss;
+        for (auto byte : data_) {
+            oss << std::hex << std::setw(2) << std::setfill('0') << static_cast<int>(byte);
+            if (spacing) {
+                oss << " ";
+            }
+        }
+        return oss.str();
+    }
+
+    /**
+     * @brief Slices  the byte array from a specified start index to an end index.
+     *
+     * This method creates a new byte array containing bytes from the specified range.
+     * The start index is inclusive, while the end index is exclusive.
+     *
+     * @param start The starting index from which to begin slicing.
+     * @param end The ending index up to which to slice (exclusive).
+     * @return byte array A new byte array containing the sliced segment.
+     * @throws std::out_of_range If the start index is greater than the end index or if the end index is out of bounds.
+     */
+    bytearray slice(size_t start, size_t end) const {
+        if (start > end || end > data_.size()) {
+            throw std::out_of_range("Invalid slice range");
+        }
+        return bytearray(std::vector<uint8_t>(data_.begin() + start, data_.begin() + end));
+    }
+
+    /**
+     * @brief Slices the byte array from a specified start index to the end of the array.
+     *
+     * This method creates a new byte array containing all bytes from the specified start index to the end of the
+     * byte array.
+     *
+     * @param start The starting index from which to begin slicing.
+     * @return byte array A new byte array containing the sliced segment from the start index to the end.
+     * @throws std::out_of_range If the start index is out of the bounds of the byte array.
+     */
+    bytearray slice_from(size_t start) const { return slice(start, data_.size()); }
+
+    /**
+     * @brief Slices the byte array from the beginning to a specified end index.
+     *
+     * This method creates a new byte array containing all bytes from the beginning of the byte array to the specified
+     * end index.
+     *
+     * @param end The ending index up to which to slice (exclusive).
+     * @return byte array A new byte array containing the sliced segment from the beginning to the end index.
+     * @throws std::out_of_range If the end index is out of the bounds of the byte array.
+     */
+    bytearray slice_to(size_t end) const { return slice(0, end); }
+
+    /**
+     * @brief Overloaded stream insertion operator for byte array.
+     * @param os The output stream.
+     * @param byteArray The byte array object.
+     * @return The output stream.
+     */
+    friend std::ostream& operator<<(std::ostream& os, const bytearray& byteArray) {
+        os << byteArray.toHex(true);
+        return os;
+    }
+
+    /**
+     * @brief Conversion operator to convert byte array to std::string.
+     *
+     * @note This is provided to support code that relies on byte array
+     *       being representd as a string.
+     * @return String containing the raw bytes of the byte array
+     */
+    operator std::string() const { return std::string(data_.begin(), data_.end()); }
+
+    //! @cond Doxygen_Suppress
+    // Expose vector-like functionality
+    size_t size() const { return data_.size(); }
+    const uint8_t* data() const { return data_.data(); }
+    bool empty() const { return data_.empty(); }
+    void clear() { data_.clear(); }
+    uint8_t& operator[](size_t index) { return data_[index]; }
+    const uint8_t& operator[](size_t index) const { return data_[index]; }
+    void push_back(uint8_t byte) { data_.push_back(byte); }
+    auto begin() const { return data_.begin(); }
+    auto end() const { return data_.end(); }
+    //! @endcond
+
+  private:
+    std::vector<uint8_t> data_;
+};
+
+}  // namespace kvn
+
+#endif  // KVN_BYTEARRAY_H

--- a/src/thirdparty/simplebluez/interfaces/Device1.h
+++ b/src/thirdparty/simplebluez/interfaces/Device1.h
@@ -5,6 +5,8 @@
 
 #include <string>
 
+#include "simplebluez/Types.h"
+
 namespace SimpleBluez {
 
 class Device1 : public SimpleDBus::Interface {
@@ -27,8 +29,8 @@ class Device1 : public SimpleDBus::Interface {
     std::string Alias();
     std::string Name();
     std::vector<std::string> UUIDs();
-    std::map<uint16_t, std::vector<uint8_t>> ManufacturerData(bool refresh = true);
-    std::map<std::string, std::vector<uint8_t>> ServiceData(bool refresh = true);
+    std::map<uint16_t, ByteArray> ManufacturerData(bool refresh = true);
+    std::map<std::string, ByteArray> ServiceData(bool refresh = true);
     bool Paired(bool refresh = true);
     bool Connected(bool refresh = true);
     bool ServicesResolved(bool refresh = true);
@@ -48,8 +50,8 @@ class Device1 : public SimpleDBus::Interface {
     std::string _address_type;
     bool _connected;
     bool _services_resolved;
-    std::map<uint16_t, std::vector<uint8_t>> _manufacturer_data;
-    std::map<std::string, std::vector<uint8_t>> _service_data;
+    std::map<uint16_t, ByteArray> _manufacturer_data;
+    std::map<std::string, ByteArray> _service_data;
 };
 
 }  // namespace SimpleBluez

--- a/src/thirdparty/simpledbus/advanced/Proxy.h
+++ b/src/thirdparty/simpledbus/advanced/Proxy.h
@@ -2,6 +2,7 @@
 
 #include <simpledbus/advanced/Interface.h>
 #include <simpledbus/external/kvn_safe_callback.hpp>
+#include <simpledbus/base/Path.h>
 
 #include <memory>
 #include <mutex>
@@ -59,6 +60,19 @@ class Proxy {
         std::scoped_lock lock(_child_access_mutex);
         for (auto& [path, child] : _children) {
             result.push_back(std::dynamic_pointer_cast<T>(child));
+        }
+        return result;
+    }
+
+    template <typename T>
+    std::vector<std::shared_ptr<T>> children_casted_with_prefix(const std::string& prefix) {
+        std::vector<std::shared_ptr<T>> result;
+        std::scoped_lock lock(_child_access_mutex);
+        for (auto& [path, child] : _children) {
+            const std::string next_child = SimpleDBus::Path::next_child_strip(_path, path);
+            if (next_child.find(prefix) == 0) {
+                result.push_back(std::dynamic_pointer_cast<T>(child));
+            }
         }
         return result;
     }

--- a/src/thirdparty/simpledbus/base/Holder.h
+++ b/src/thirdparty/simpledbus/base/Holder.h
@@ -8,6 +8,28 @@
 
 namespace SimpleDBus {
 
+class ObjectPath {
+  public:
+    ObjectPath(const std::string& path) : path(path) {}
+    ObjectPath(const char* path) : path(path) {}
+    operator std::string() const { return path; }
+    bool operator<(const ObjectPath& other) const { return path < other.path; }
+
+  private:
+    std::string path;
+};
+
+class Signature {
+  public:
+    Signature(const std::string& signature) : signature(signature) {}
+    Signature(const char* signature) : signature(signature) {}
+    operator std::string() const { return signature; }
+    bool operator<(const Signature& other) const { return signature < other.signature; }
+
+  private:
+    std::string signature;
+};
+
 class Holder;
 
 class Holder {
@@ -37,9 +59,10 @@ class Holder {
     } Type;
 
     Type type() const;
-    std::string represent();
-    std::string signature();
+    std::string represent() const;
+    std::string signature() const;
 
+    // TODO: Deprecate these functions in favor of templated version.
     static Holder create_boolean(bool value);
     static Holder create_byte(uint8_t value);
     static Holder create_int16(int16_t value);
@@ -50,12 +73,14 @@ class Holder {
     static Holder create_uint64(uint64_t value);
     static Holder create_double(double value);
     static Holder create_string(const std::string& str);
-    static Holder create_object_path(const std::string& str);
-    static Holder create_signature(const std::string& str);
+    static Holder create_object_path(const ObjectPath& path);
+    static Holder create_signature(const Signature& signature);
     static Holder create_array();
     static Holder create_dict();
 
     std::any get_contents() const;
+
+    // TODO: Deprecate these functions in favor of templated version.
     bool get_boolean() const;
     uint8_t get_byte() const;
     int16_t get_int16() const;
@@ -83,6 +108,16 @@ class Holder {
     void dict_append(Type key_type, std::any key, Holder value);
     void array_append(Holder holder);
 
+    // Template speciallizations.
+    template <typename T>
+    static Holder create();
+
+    template <typename T>
+    static Holder create(T value);
+
+    template <typename T>
+    T get() const;
+
   private:
     Type _type = NONE;
 
@@ -96,15 +131,15 @@ class Holder {
     // Dictionaries are stored within a vector as a tuple of <key_type, key, holder>
     std::vector<std::tuple<Type, std::any, Holder>> holder_dict;
 
-    std::vector<std::string> _represent_container();
-    std::string _represent_simple();
-    std::string _signature_simple();
+    std::vector<std::string> _represent_container() const;
+    std::string _represent_simple() const;
+    std::string _signature_simple() const;
 
     template <typename T>
     std::map<T, Holder> _get_dict(Type key_type) const;
 
-    static std::string _signature_type(Type type);
-    static std::string _represent_type(Type type, std::any value);
+    static std::string _signature_type(Type type) noexcept;
+    static std::string _represent_type(Type type, std::any value) noexcept;
 };
 
 }  // namespace SimpleDBus

--- a/src/thirdparty/simpledbus/base/Message.h
+++ b/src/thirdparty/simpledbus/base/Message.h
@@ -1,76 +1,73 @@
 #pragma once
 
 #include <dbus/dbus.h>
-
 #include <atomic>
-#include <stack>
 #include <string>
 #include <vector>
-
-#include "Connection.h"
 #include "Holder.h"
 
 namespace SimpleDBus {
 
-class Connection;
-class Interface;
-
 class Message {
   public:
-    typedef enum {
+    enum class Type {
         INVALID = DBUS_MESSAGE_TYPE_INVALID,
         METHOD_CALL = DBUS_MESSAGE_TYPE_METHOD_CALL,
         METHOD_RETURN = DBUS_MESSAGE_TYPE_METHOD_RETURN,
         ERROR = DBUS_MESSAGE_TYPE_ERROR,
         SIGNAL = DBUS_MESSAGE_TYPE_SIGNAL,
-    } Type;
+    };
+
+    static constexpr int INVALID_UNIQUE_ID = -1;
 
     Message();
-    Message(DBusMessage* msg);
-    Message(Message&& other);                  // Custom move constructor
-    Message(const Message& other);             // Custom copy constructor
-    Message& operator=(Message&& other);       // Custom move assignment
-    Message& operator=(const Message& other);  // Custom copy assignment
+    Message(Message&& other) noexcept;
+    Message(const Message& other);
+    Message& operator=(Message&& other) noexcept;
+    Message& operator=(const Message& other);
     ~Message();
 
-    bool is_valid() const;
-    void append_argument(Holder argument, std::string signature);
+    operator DBusMessage*() const;
 
+    bool is_valid() const;
+    void append_argument(const Holder& argument, const std::string& signature);
     Holder extract();
     void extract_reset();
     bool extract_has_next();
     void extract_next();
     std::string to_string(bool append_arguments = false) const;
 
-    int32_t get_unique_id();
-    uint32_t get_serial();
+    uint32_t get_ref_count() const;
+    int32_t get_unique_id() const;
+    uint32_t get_serial() const;
     std::string get_signature();
-    std::string get_interface();
-    std::string get_path();
-    std::string get_member();
+    std::string get_interface() const;
+    std::string get_path() const;
+    std::string get_member() const;
     Type get_type() const;
 
-    bool is_signal(std::string interface, std::string signal_name);
+    bool is_signal(const std::string& interface, const std::string& signal_name) const;
+    bool is_method_call(const std::string& interface, const std::string& method) const;
 
-    static Message create_method_call(std::string bus_name, std::string path, std::string interface,
-                                      std::string method);
-
+    static Message from_retained(DBusMessage* msg);
+    static Message from_acquired(DBusMessage* msg);
+    static Message create_method_call(const std::string& bus_name, const std::string& path,
+                                      const std::string& interface, const std::string& method);
     static Message create_method_return(const Message& msg);
-
-    static Message create_error(const Message& msg, std::string error_name, std::string error_message);
+    static Message create_error(const Message& msg, const std::string& error_name, const std::string& error_message);
+    static Message create_signal(const std::string& path, const std::string& interface, const std::string& signal);
 
   private:
-    friend class Connection;
+    static std::atomic_int32_t _creation_counter;
 
-    static std::atomic_int32_t creation_counter;
-    int indent;
-
-    int _unique_id;
+    int _indent = 0;
+    int32_t _unique_id = INVALID_UNIQUE_ID;
     DBusMessageIter _iter;
-    bool _iter_initialized;
-    bool _is_extracted;
+    bool _iter_initialized = false;
+    bool _is_extracted = false;
     Holder _extracted;
-    DBusMessage* _msg;
+    DBusMessage* _msg = nullptr;
+    std::vector<Holder> _arguments;
 
     Holder _extract_bytearray(DBusMessageIter* iter);
     Holder _extract_array(DBusMessageIter* iter);
@@ -83,12 +80,10 @@ class Message {
      * @param argument  Argument to append.
      * @param signature Signature of the argument.
      */
-    void _append_argument(DBusMessageIter* iter, Holder& argument, std::string signature);
+    void _append_argument(DBusMessageIter* iter, const Holder& argument, const std::string& signature);
 
     void _invalidate();
     void _safe_delete();
-
-    std::vector<Holder> _arguments;
 };
 
 }  // namespace SimpleDBus

--- a/src/thirdparty/simpledbus/base/Path.h
+++ b/src/thirdparty/simpledbus/base/Path.h
@@ -18,6 +18,7 @@ class Path {
     static bool is_parent(const std::string& base, const std::string& path);
 
     static std::string next_child(const std::string& base, const std::string& path);
+    static std::string next_child_strip(const std::string& base, const std::string& path);
 };
 
 }  // namespace SimpleDBus

